source/material_model/rheology/strain_dependent.cc

```diff
@@ -360,6 +404,32 @@ namespace aspect
 
       }
 
+      template <int dim>
+      double
+      StrainDependent<dim>::
+      calculate_strain_healing(const MaterialModel::MaterialModelInputs<dim> &in,
+                               const double edot_ii,
+                               const unsigned int j) const
+      {
+        const double reference_temperature = this->get_adiabatic_surface_temperature();
+        double healed_strain = edot_ii * this->get_timestep();
+
+        switch (healing_mechanism)
+          {
+            case no_healing:
+            {
+              break;
+            }
+            case temperature_dependent:
+            {
+              healed_strain -= in.composition[j][0] * strain_healing_recovery_timescale *
```

Comment:
There is an error in this line. `in.composition[j][0]` is just accessing the first compositional field, which may or may not be a strain value. I'm actually not sure why you need the composition term in this equation?

Reply:
I was thinking about that, I think its just an artifact of how it was written in the older version of the code

Reply:
In that case can you remove `MaterialModel::MaterialModelInputs<dim> &in` and `j` as inputs to `calculate_strain_healing`?

Reply:
Wouldn't I still need those to access the temperature at the point?

Reply:
yes, of course. No need to modify the interface.